core/trino-main/src/main/java/io/trino/operator/TableUpdateOperator.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator;
+
+import io.trino.Session;
+import io.trino.metadata.Metadata;
+import io.trino.metadata.TableHandle;
+import io.trino.spi.Page;
+import io.trino.sql.planner.plan.PlanNodeId;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class TableUpdateOperator
+        implements Operator
+{
+    private final TableMetadataOperator metadataOperator;
+
+    public static class TableUpdateOperatorFactory
+            implements OperatorFactory
+    {
+        private final int operatorId;
+        private final PlanNodeId planNodeId;
+        private final Metadata metadata;
+        private final Session session;
+        private final TableHandle tableHandle;
+        private boolean closed;
+
+        public TableUpdateOperatorFactory(int operatorId, PlanNodeId planNodeId, Metadata metadata, Session session, TableHandle tableHandle)
+        {
+            this.operatorId = operatorId;
+            this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
+            this.metadata = requireNonNull(metadata, "metadata is null");
+            this.session = requireNonNull(session, "session is null");
+            this.tableHandle = requireNonNull(tableHandle, "tableHandle is null");
+        }
+
+        @Override
+        public Operator createOperator(DriverContext driverContext)
+        {
+            checkState(!closed, "Factory is already closed");
+            OperatorContext context = driverContext.addOperatorContext(operatorId, planNodeId, TableUpdateOperator.class.getSimpleName());
+            return new TableUpdateOperator(context, metadata, session, tableHandle);
+        }
+
+        @Override
+        public void noMoreOperators()
+        {
+            closed = true;
+        }
+
+        @Override
+        public OperatorFactory duplicate()
+        {
+            return new TableUpdateOperatorFactory(operatorId, planNodeId, metadata, session, tableHandle);
+        }
+    }
+
+    public TableUpdateOperator(OperatorContext operatorContext, Metadata metadata, Session session, TableHandle tableHandle)
+    {
+        this.metadataOperator = new TableMetadataOperator(operatorContext, () -> metadata.executeUpdate(session, tableHandle));
```

Comment:
The only difference between this class and TableDeleteOperator is the lambda expression in this line. Therefore, the TableUpdateOperator and TableDeleteOperators are unnecessary. They just add complexity.

Instead, the code could just build the OperatorFactory with the lambda directly in LocalExecutionPlanner.

Reply:
refactored